### PR TITLE
Add rvv implementation for unary/binary vector.

### PR DESCRIPTION
**Pull request by zhangyang2057**
- Support m1/m2/m4/m8.
- Add more ctest cases to cover.

### CODE REVIEW

File: src/Native/test/test_ntt_binary_max.cpp [hunk 1]
```diff
@@ -709,28 +709,36 @@ TEST(BinaryTestMaxFloat, ranked_ranked_ranked_broadcast_multidirectional) {
...
+    NttTest::init_tensor(ntt_lhs, static_cast<T>(-10), static_cast<T>(10));
+    NttTest::init_tensor(ntt_rhs, static_cast<T>(-10), static_cast<T>(10));
     auto ntt_output1 = ntt::max(ntt_lhs, ntt_rhs);
-
-    // ort
     auto ort_lhs = NttTest::ntt2ort(ntt_lhs);
     auto ort_rhs = NttTest::ntt2ort(ntt_rhs);
     ortki::OrtKITensor *ort_array[] = {ort_lhs, ort_rhs};
     auto ort_output =
         ortki_Max(ort_array, sizeof(ort_array) / sizeof(ort_array[0]));
```
Comment (sunnycase) [inline src/Native/test/test_ntt_binary_max.cpp:721]: It's better to use `std::size` to replace `sizeof(ort_array) / sizeof(ort_array[0])`

File: src/Native/test/test_ntt_binary_min.cpp [hunk 2]
```diff
@@ -709,28 +709,36 @@ TEST(BinaryTestMinFloat, ranked_ranked_ranked_broadcast_multidirectional) {
...
+    NttTest::init_tensor(ntt_lhs, static_cast<T>(-10), static_cast<T>(10));
+    NttTest::init_tensor(ntt_rhs, static_cast<T>(-10), static_cast<T>(10));
     auto ntt_output1 = ntt::min(ntt_lhs, ntt_rhs);
-
-    // ort
     auto ort_lhs = NttTest::ntt2ort(ntt_lhs);
     auto ort_rhs = NttTest::ntt2ort(ntt_rhs);
     ortki::OrtKITensor *ort_array[] = {ort_lhs, ort_rhs};
     auto ort_output =
         ortki_Min(ort_array, sizeof(ort_array) / sizeof(ort_array[0]));
```
Comment (sunnycase) [inline src/Native/test/test_ntt_binary_min.cpp:721]: Same as above.

File: src/Native/test/test_ntt_unary_swish.cpp [hunk 3]
```diff
@@ -140,32 +140,39 @@ TEST(UnaryTestSwishFloat, ranked_fixed) {
...
     auto ort_input = NttTest::ntt2ort(ntt_input);
-    float data[1] = {1.f};
+    T data[1] = {static_cast<T>(1)};
     int64_t one_shape[1] = {1};
-    auto one_tensor =
-        make_tensor(reinterpret_cast<void *>(data), ortki::DataType_FLOAT,
-                    one_shape, sizeof(one_shape) / sizeof(one_shape[0]));
+    auto one_tensor = make_tensor(
+        reinterpret_cast<void *>(data), NttTest::primitive_type2ort_type<T>(),
+        one_shape, sizeof(one_shape) / sizeof(one_shape[0]));
```
Comment (zhangyang2057) [inline src/Native/test/test_ntt_unary_swish.cpp:152]: Use std::size(one_shape) instead.

File: src/Native/test/test_ntt_binary_mod.cpp [hunk 4]
```diff
@@ -684,67 +684,35 @@ TEST(BinaryTestModInt32, fixed_fixed_fixed) {
...
-    // ort
-    auto ort_lhs = NttTest::ntt2ort(ntt_lhs);
-    auto ort_rhs = NttTest::ntt2ort(ntt_rhs);
-    auto ort_output = ortki_Mod(ort_lhs, ort_rhs, 1);
-
-    // compare
-    ntt::vector<double, 8> ntt_output2;
-    NttTest::ort2ntt(ort_output, ntt_output2);
-    EXPECT_TRUE(NttTest::compare_tensor(ntt_output1, ntt_output2));
+TEST(UnaryTestPow, vector) {
```
Comment (zhangyang2057) [inline src/Native/test/test_ntt_binary_mod.cpp:709]: Mod

File: src/Native/test/test_ntt_unary_tanh.cpp [hunk 5]
```diff
@@ -108,24 +108,28 @@ TEST(UnaryTestTanhFloat, ranked_fixed) {
...
+#define _TEST_VECTOR(T, lmul)                                                  \
+    test_vector<T, (NTT_VLEN) / (sizeof(T) * 8) * lmul>();
+
+#define TEST_VECTOR(T)                                                         \
+    _TEST_VECTOR(T, 1)                                                         \
+    _TEST_VECTOR(T, 2)                                                         \
+    _TEST_VECTOR(T, 4)                                                         \
+    _TEST_VECTOR(T, 8)
+
+TEST(UnaryTesttanh, vector) { TEST_VECTOR(float) }
```
Comment (zhangyang2057) [inline src/Native/test/test_ntt_unary_tanh.cpp:131]: Tanh

**Review by sunnycase — APPROVED** [review]

**Pull request merged**